src/Utility/Import.php

```diff
@@ -0,0 +1,309 @@
+        'defaults' => [
+            'status' => 'on',
+        ],
+        'csv' => [
+            'delimiter' => ',',
+            'enclosure' => '"',
+            'escape' => '"',
+        ],
+    ];
+
+    /**
+     * Dry run mode flag
+     *
+     * @var bool
+     */
+    public bool $dryrun = false;
+
+    /**
+     * Full filename path
+     *
+     * @var string|null
+     */
+    public ?string $filename = '';
+
+    /**
+     * Parent uname or ID
+     *
+     * @var string|null
+     */
+    public ?string $parent = '';
+
+    /**
+     * Number of processed entities
+     *
+     * @var int
+     */
+    public int $processed = 0;
+
+    /**
+     * Number of saved entities
+     *
+     * @var int
+     */
+    public int $saved = 0;
+
+    /**
+     * Number of errors
+     *
+     * @var int
+     */
+    public int $errors = 0;
+
+    /**
+     * Errors details
+     *
+     * @var array
+     */
+    public array $errorsDetails = [];
+
+    /**
+     * Number of skipped
+     *
+     * @var int
+     */
+    public int $skipped = 0;
+
+    /**
+     * Entity type
+     *
+     * @var string
+     */
+    public string $type = '';
+
+    /**
+     * Objects table
+     *
+     * @var \BEdita\Core\Model\Table\ObjectsTable
+     */
+    protected $objectsTable;
+
+    /**
+     * Type table
+     *
+     * @var \BEdita\Core\Model\Table\ObjectsTable
+     */
+    protected $typeTable;
+
+    /**
+     * Translations table
+     *
+     * @var \BEdita\Core\Model\Table\TranslationsTable
+     */
+    protected $translationsTable;
+
+    /**
+     * Constructor
+     *
+     * @param string|null $filename Full filename path
+     * @param string|null $type Entity type
+     * @param string|null $parent Parent uname or ID
+     * @param bool|null $dryrun Dry run mode flag
+     * @return void
+     */
+    public function __construct(
+        ?string $filename = null,
+        ?string $type = 'objects',
+        ?string $parent = null,
+        ?bool $dryrun = false
+    ) {
+        $this->filename = $filename;
+        $this->type = $type;
+        $this->parent = $parent;
+        $this->dryrun = $dryrun;
+        $this->processed = 0;
+        $this->saved = 0;
+        $this->errors = 0;
+        $this->skipped = 0;
+        $this->errorsDetails = [];
+        /** @var \BEdita\Core\Model\Table\ObjectsTable $objectsTable */
+        $objectsTable = $this->fetchTable('objects');
+        $this->objectsTable = $objectsTable;
+        /** @var \BEdita\Core\Model\Table\ObjectsTable $typesTable */
+        $typesTable = $this->fetchTable($this->type);
+        $this->typeTable = $typesTable;
+        /** @var \BEdita\Core\Model\Table\TranslationsTable $translationsTable */
+        $translationsTable = $this->fetchTable('translations');
+        $this->translationsTable = $translationsTable;
+    }
+
+    /**
+     * Save objects
+     *
+     * @return void
+     */
+    public function objects(): void
```

Comment:
I'd use `saveObjects()`, `saveObject()`, `saveTranslation()`, `saveTranslations()` as method names for save actions

Reply:
good point. I'd apply a refactor with this suggestion